src/lib/y2network/dialogs/rename_interface.rb

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "cwm/popup"
+require "y2network/widgets/custom_interface_name"
+require "y2network/widgets/rename_hwinfo"
+
+module Y2Network
+  module Dialogs
+    # This dialog allows the user to rename a network interface
+    #
+    # Is works in a slightly different way depending on the interface.
+    #
+    # * For physical interfaces, it allows the user to select between using
+    #   the MAC adddress or the Bus ID, which are present in the Hwinfo object
+    #   associated to the interface.
+    # * For not connected interfaces ({FakeInterface}), as the hardware is not present,
+    #   the user must specify the MAC or the Bus ID by hand (NOT IMPLEMENTED YET).
+    # * For virtual interfaces, like bridges, only the name can be chaned (no hardware
```

Comment:
do we need for this third case this dialog at all? I though that simply allow to specify name in general tab is enough, not?

Reply:
I prefer to use the same workflow for all renames. It should be easy to change, anyway.

Reply:
Updated. Now the dialog is only shown when the required hardware information is available.